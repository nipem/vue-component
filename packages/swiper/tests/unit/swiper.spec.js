/* eslint-disable */

import {
  mount,
  shallowMount
} from '@vue/test-utils';
import {
  Swiper,
  Slide
} from '../../lib/index';

describe('Swiper component', () => {
  describe('Default mounting properties', () => {
    it('should mount successfully', () => {
      const wrapper = mount(Swiper);

      expect(wrapper.vm._isMounted).toBeTruthy();

      expect(wrapper).toMatchSnapshot();
    });

    it('should unmount successfully', () => {
      const wrapper = mount(Swiper);

      const SwiperInstance = wrapper.vm;
      SwiperInstance.$destroy();
      expect(SwiperInstance._isDestroyed).toBeTruthy();

      expect(wrapper).toMatchSnapshot();
    });

    it('should be able to autoplay by default', () => {
      const wrapper = mount(Swiper);

      expect(wrapper.vm.autoplay).toBeTruthy();

      expect(wrapper).toMatchSnapshot();
    });

    it('should be able to loop by default', () => {
      const wrapper = mount(Swiper);

      expect(wrapper.vm.loop).toBeTruthy();

      expect(wrapper).toMatchSnapshot();
    });

    it('should have 3 items when number of slides is 3', done => {
      const wrapper = mount(Swiper, {
        slots: {
          default: [Slide, Slide, Slide]
        }
      });

      expect(wrapper.vm.items.length).toBe(3);

      wrapper.vm.$nextTick(() => {
        expect(wrapper).toMatchSnapshot();
        done();
      });
    });
  });

  describe('Events', () => {
    it('should activeIndex be 2 when emit next event twice', done => {
      const wrapper = shallowMount(Swiper, {
        slots: {
          default: [Slide, Slide, Slide]
        }
      });
      wrapper.vm.next();
      wrapper.vm.next();

      expect(wrapper.vm.activeIndex).toBe(2);

      wrapper.vm.$nextTick(() => {
        expect(wrapper).toMatchSnapshot();
        done();
      });
    });

    it('should activeIndex be 0 when emit next at the last slide', done => {
      const wrapper = shallowMount(Swiper, {
        slots: {
          default: [Slide, Slide, Slide]
        },
        propsData: {
          active: 2
        }
      });
      wrapper.vm.next();

      expect(wrapper.vm.activeIndex).toBe(0);

      wrapper.vm.$nextTick(() => {
        expect(wrapper).toMatchSnapshot();
        done();
      });
    });

    it('should activeIndex be 2 when emit prev at the first slide', done => {
      const wrapper = shallowMount(Swiper, {
        slots: {
          default: [Slide, Slide, Slide]
        },
        propsData: {
          active: 0
        }
      });
      wrapper.vm.prev();

      expect(wrapper.vm.activeIndex).toBe(2);

      wrapper.vm.$nextTick(() => {
        expect(wrapper).toMatchSnapshot();
        done();
      });
    });

    it('should activeIndex be 2 when emit changeActiveIndex(2)', done => {
      const wrapper = shallowMount(Swiper, {
        slots: {
          default: [Slide, Slide, Slide]
        },
      });
      wrapper.vm.changeActiveIndex(2);

      expect(wrapper.vm.activeIndex).toBe(2);

      wrapper.vm.$nextTick(() => {
        expect(wrapper).toMatchSnapshot();
        done();
      });
    });
  });
});
