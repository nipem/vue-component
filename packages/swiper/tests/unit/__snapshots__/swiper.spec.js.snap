// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Swiper component Default mounting properties should be able to autoplay by default 1`] = `
<div class="swiper">
  <div class="swiper-wrapper"></div>
  <!---->
  <!---->
</div>
`;

exports[`Swiper component Default mounting properties should be able to loop by default 1`] = `
<div class="swiper">
  <div class="swiper-wrapper"></div>
  <!---->
  <!---->
</div>
`;

exports[`Swiper component Default mounting properties should have 3 items when number of slides is 3 1`] = `
<div class="swiper">
  <div class="swiper-wrapper">
    <div class="slide-item" enter-class="" enter-active-class="animated fadeIn" enter-to-class="" leave-class="" leave-active-class="animated fadeOut" leave-to-class=""></div>
    <div class="slide-item" style="display: none;" enter-class="" enter-active-class="animated fadeIn" enter-to-class="" leave-class="" leave-active-class="animated fadeOut" leave-to-class=""></div>
    <div class="slide-item" style="display: none;" enter-class="" enter-active-class="animated fadeIn" enter-to-class="" leave-class="" leave-active-class="animated fadeOut" leave-to-class=""></div>
  </div>
  <!---->
  <!---->
</div>
`;

exports[`Swiper component Default mounting properties should mount successfully 1`] = `
<div class="swiper">
  <div class="swiper-wrapper"></div>
  <!---->
  <!---->
</div>
`;

exports[`Swiper component Default mounting properties should unmount successfully 1`] = `
<div class="swiper">
  <div class="swiper-wrapper"></div>
  <!---->
  <!---->
</div>
`;

exports[`Swiper component Events should activeIndex be 0 when emit next at the last slide 1`] = `
<div class="swiper">
  <div class="swiper-wrapper">
    <div class="slide-item" style="" enter-class="" enter-active-class="animated fadeIn" enter-to-class="" leave-class="" leave-active-class="animated fadeOut" leave-to-class=""></div>
    <div class="slide-item" style="display: none;" enter-class="" enter-active-class="animated fadeIn" enter-to-class="" leave-class="" leave-active-class="animated fadeOut" leave-to-class=""></div>
    <div class="slide-item" enter-class="" enter-active-class="animated fadeIn" enter-to-class="" leave-class="" leave-active-class="animated fadeOut" leave-to-class="" style="display: none;"></div>
  </div>
  <!---->
  <!---->
</div>
`;

exports[`Swiper component Events should activeIndex be 2 when emit changeActiveIndex(2) 1`] = `
<div class="swiper">
  <div class="swiper-wrapper">
    <div class="slide-item" enter-class="" enter-active-class="animated fadeIn" enter-to-class="" leave-class="" leave-active-class="animated fadeOut" leave-to-class="" style="display: none;"></div>
    <div class="slide-item" style="display: none;" enter-class="" enter-active-class="animated fadeIn" enter-to-class="" leave-class="" leave-active-class="animated fadeOut" leave-to-class=""></div>
    <div class="slide-item" style="" enter-class="" enter-active-class="animated fadeIn" enter-to-class="" leave-class="" leave-active-class="animated fadeOut" leave-to-class=""></div>
  </div>
  <!---->
  <!---->
</div>
`;

exports[`Swiper component Events should activeIndex be 2 when emit next event twice 1`] = `
<div class="swiper">
  <div class="swiper-wrapper">
    <div class="slide-item" enter-class="" enter-active-class="animated fadeIn" enter-to-class="" leave-class="" leave-active-class="animated fadeOut" leave-to-class="" style="display: none;"></div>
    <div class="slide-item" style="display: none;" enter-class="" enter-active-class="animated fadeIn" enter-to-class="" leave-class="" leave-active-class="animated fadeOut" leave-to-class=""></div>
    <div class="slide-item" style="" enter-class="" enter-active-class="animated fadeIn" enter-to-class="" leave-class="" leave-active-class="animated fadeOut" leave-to-class=""></div>
  </div>
  <!---->
  <!---->
</div>
`;

exports[`Swiper component Events should activeIndex be 2 when emit prev at the first slide 1`] = `
<div class="swiper">
  <div class="swiper-wrapper">
    <div class="slide-item" enter-class="" enter-active-class="animated fadeIn" enter-to-class="" leave-class="" leave-active-class="animated fadeOut" leave-to-class="" style="display: none;"></div>
    <div class="slide-item" style="display: none;" enter-class="" enter-active-class="animated fadeIn" enter-to-class="" leave-class="" leave-active-class="animated fadeOut" leave-to-class=""></div>
    <div class="slide-item" style="" enter-class="" enter-active-class="animated fadeIn" enter-to-class="" leave-class="" leave-active-class="animated fadeOut" leave-to-class=""></div>
  </div>
  <!---->
  <!---->
</div>
`;
